### ISSUE TITLE
Deploying of Drupal multisite on Acquia Cloud fails is sites use aliases.

### ISSUE THREAD

**Issue by mirsoftacquia** [acquia/blt#3200]
My system information:
* Operating system type: Max OSX 
* Operating system version: 10.13.6
* BLT version: 9.2.0

Standard Drupal 8 multisite, each site uses different local alias, e.g. @site1.local , @site2.local etc. After deployment to Acquia Cloud, deployment fails with following error:

```
blt artifact:ac-hooks:post-code-update $site $target_env $source_branch $deployed_tag $repo_url $repo_type --environment=$target_env -v --yes --no-interaction
[warning] Drupal VM is locally initialized, but is not running.
Running updates for environment: dev
> artifact:update:drupal:all-sites
Deploying updates to default...
> drupal:config:import
Executing pre-config-import target hook...
[ExecStack] echo 'No pre-config-import configured.'
[ExecStack] Running echo 'No pre-config-import configured.' in /var/www/html/mysite.dev/docroot
No pre-config-import configured.
[ExecStack] Done in 0.002s
[Acquia\Blt\Robo\Tasks\DrushTask] Running /var/www/html/mysite.dev/vendor/bin/drush @site1.local cc drush --no-interaction -v --ansi in /var/www/html/mysite.dev/docroot
 [37;46;1m[success][39;49;22m 'drush' cache was cleared.

[Acquia\Blt\Robo\Tasks\DrushTask] Done in 0.419s
[Acquia\Blt\Robo\Tasks\DrushTask] Running /var/www/html/mysite.dev/vendor/bin/drush @site1.local cache-rebuild --no-interaction -v --ansi in /var/www/html/mysite.dev/docroot

[33mIn BootstrapHook.php line 32:[39m
[37;41m                                                                      [39;49m
[37;41m  [Exception]                                                         [39;49m
[37;41m  Bootstrap failed. Run your command with -vvv for more information.  [39;49m
[37;41m                                                                      [39;49m
```

After digging into the problem deeper, it seems the problem is in usage of @site1.local alias that should not be used on artifact:ac-hooks:post-code-update at all. In drush9, these commands should be probably invoked without alias.

Looking into BLT's `DeployCommand.php` there is really command `$this->config->set('drush.alias', '');` that clears these aliases on update. However, after that, the `$this->switchSiteContext($multisite);` command is invoked in `updateSite()` method that sets the alias back to the local site's alias.

The fix that worked for me was to move the `$this->config->set('drush.alias', '');` command after context switch. The corrected drush command looked then like:

```
Running /var/www/html/mysite.dev/vendor/bin/drush cache-rebuild --no-interaction -v --ansi in /var/www/html/mysite.dev/docroot
```

The PR follows soon.

**Comment by mikemadison13** [comment]
@mirsoftacquia does @lcatlett's implementation by chance address this issue in a different way? https://github.com/acquia/blt/pull/3231

**Comment by lcatlett** [comment]
@mirsoftacquia this was fixed in #3231, https://github.com/acquia/blt/commit/1d2f22be4458b61b471ef97c463373040ef042cf#diff-3f01be0f15533ad5bd0f496e6bec3e74 specifically the `$site_path` config and the updated Acquia host detection. Additionally, there are a few other bugs in the execution of your Cloud Hooks and deploy config imports that were fixed in https://github.com/acquia/blt/commit/2d730914b018f5558fae05f1cf233904cc0e3181#diff-3d8cbed0246f21f2e5dee994d04cb0f6. The cr and cc are not only redundant but a common cause of drush losing the site context during deploys. Closing this out since #3231 was merged
